snakemake/workflow.py

```diff
@@ -54,6 +54,7 @@
 from snakemake.persistence import Persistence
 from snakemake.utils import update_config
 from snakemake.script import script
+from snakemake.notebook import notebook
```

Comment:
May I ask you to now move the Notebook specific implementations of the ScriptBase class into a new module `snakemake.notebook` and add a function `notebook(...)` there, similar to the `script(...)` function in `snakemake.script`. Also, could you please move the script specific implementations of ScriptBase into the `snakemake.script` module?